### PR TITLE
Update sample and explain about TLS best practices

### LINKED ISSUE #12224: The sample doesn't work
I've downloaded the sample and am trying to run the QueueResults app.

When pressing the Start button it writes "Downloads failed." (Actually there's a HttpRequestException: An error occurred while sending the request.)

And the following messages are in the output:
	Exception thrown: 'System.Security.Authentication.AuthenticationException' in System.dll
	Exception thrown: 'System.ObjectDisposedException' in System.dll
	Exception thrown: 'System.Net.WebException' in System.dll
	Exception thrown: 'System.Net.Http.HttpRequestException' in mscorlib.dll

And I've tried other samples in "The Task asynchronous programming model in C#" tutorial and none of them work.

I'm using a firewall and have opened it for the app.

And I've tried to connect to the same url via HttpClient.GetByteArrayAsync(url) in a separate test app, and it succeeds.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9695aa2e-6d63-f36f-0e8e-b552ed0bf40a
* Version Independent ID: b06fb1c9-770c-4628-516e-7d661cf0bf29
* Content: [Handling Reentrancy in Async Apps (C#)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps#feedback)
* Content Source: [docs/csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by nemrism**
## Summary

- Adding support for TLS 1.2 to make the sample work with .NET framework versions prior to 4.6
- Adding a note to explaining TLS best practices
- Applied modification to both C# and VB sections

Fixes #12224 

/cc: @BillWagner

### CODE REVIEW

File: docs/csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md [hunk 1]
```diff
@@ -603,7 +606,9 @@ The following section provides the code to build the example as a WPF app.
 
      A simple window that contains a text box and a button appears in the **Design** view of MainWindow.xaml.
 
-8. Add a reference for <xref:System.Net.Http>.
+8. In **Solution Explorer**, right-click on **References** and select **Add Reference...**.
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md]:
```suggestion
8. In **Solution Explorer**, right-click on **References** and select **Add Reference**.
```

I remember that it was mentioned somewhere not to include those dots.

File: docs/visual-basic/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md [hunk 2]
```diff
@@ -555,7 +558,7 @@ The following section provides the code to build the example as a WPF app.
 
 4. In the list of project types, choose **WPF Application**.
 
-5. Name the project `WebsiteDownloadWPF`, and then choose the **OK** button.
+5. Name the project `WebsiteDownloadWPF`, choose a .NET Framework version of `4.6+` and then the **OK** button.
```
Comment (Youssef1313) [inline docs/visual-basic/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md]:
```suggestion
5. Name the project `WebsiteDownloadWPF`, choose .NET Framework version 4.6 or higher, and then the click the **OK** button.
```

File: docs/visual-basic/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md [hunk 3]
```diff
@@ -583,7 +586,9 @@ The following section provides the code to build the example as a WPF app.
 
      A simple window that contains a text box and a button appears in the **Design** view of MainWindow.xaml.
 
-8. Add a reference for <xref:System.Net.Http>.
+8. In **Solution Explorer**, right-click on **References** and select **Add Reference...**.
```
Comment (Youssef1313) [inline docs/visual-basic/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md]:
```suggestion
8. In **Solution Explorer**, right-click on **References** and select **Add Reference**.
```

File: docs/csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md (hunk 1 above)
Comment (nemrism) [reply docs/csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md]: Ok, it makes sense. Those dots are quite useless.

Comment (Youssef1313) [reply docs/csharp/programming-guide/concepts/async/handling-reentrancy-in-async-apps.md]: Found it [here](https://docs.microsoft.com/en-us/style-guide/procedures-instructions/formatting-text-in-instructions).

>  If a dialog box label ends with a colon or an ellipsis, don't include that end punctuation in instructions.

**Review by BillWagner — APPROVED** [review]
Thanks for making these changes @nemrism We appreciate it.

I've reviewed the changes, and I'll :shipit: now.

**Pull request merged**